src/App.js

```diff
@@ -2,20 +2,29 @@ import React from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
+function useSemiPersistentState() {
+  const [todoList, setTodoList] = React.useState(
+    JSON.parse(localStorage.getItem('savedTodoList')) // read initial value of state hook form local storage
+  );
```

Comment:
Great! Just a small tip. If you set todoList equal to an empty array, the app still will work if the JSON doesn't parse correctly or there's nothing in local storage.
`JSON.parse(localStorage.getItem("savedTodoList")) || []`